src/win.rs

```diff
@@ -75,94 +85,39 @@ impl Endpoint {
 
     /// Make new connection using the provided path and running event pool.
     pub async fn connect<P: AsRef<Path>>(path: P) -> io::Result<Connection> {
-        Ok(Connection::wrap(Self::connect_inner(path.as_ref())?))
-    }
-
-    fn connect_inner(path: &Path) -> io::Result<NamedPipe> {
-        use std::fs::OpenOptions;
-        use std::os::windows::fs::OpenOptionsExt;
-        use std::os::windows::io::{FromRawHandle, IntoRawHandle};
-        use winapi::um::winbase::FILE_FLAG_OVERLAPPED;
+        let path = path.as_ref();
+
+        // There is not async equivalent of waiting for a named pipe in Windows,
+        // so we keep trying or sleeping for a bit, until we hit a timeout
+        let attempt_start = Instant::now();
+        let client = loop {
+            match named_pipe::ClientOptions::new()
+                .read(true)
+                .write(true)
+                .open(path)
+            {
+                Ok(client) => break client,
+                Err(e) if e.raw_os_error() == Some(ERROR_PIPE_BUSY as i32) => {
+                    if attempt_start.elapsed() < PIPE_AVAILABILITY_TIMEOUT {
+                        tokio::time::sleep(Duration::from_millis(50)).await;
+                        continue;
```

Comment:
This is technically not required but I wanted to emphasize that there's no fall-through: this `match` should be the only expression executed in the `loop`